tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_matrix_inf_norm_kk.hpp

```diff
@@ -0,0 +1,52 @@
+
+#ifndef LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_MATRIX_INF_NORM_HPP_
+#define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_MATRIX_INF_NORM_HPP_
+
+#include "signal_kokkos_impl_called.hpp"
+
+namespace KokkosKernelsSTD {
+
+template<
+    class ExeSpace,
+    class ElementType,
+    std::experimental::extents<>::size_type numRows,
+    std::experimental::extents<>::size_type numCols,
+    class Layout,
+    class Scalar>
+Scalar matrix_inf_norm(kokkos_exec<ExeSpace> /*kexe*/,
+			std::experimental::mdspan<
+			ElementType,
+			std::experimental::extents<numRows, numCols>,
+			Layout,
+			std::experimental::default_accessor<ElementType>> A,
+			Scalar init)
+{
+
+  Impl::signal_kokkos_impl_called("matrix_inf_norm");
+
+  if (A.extent(0) == 0){
+    return init;
+  }
+
+  auto A_view = Impl::mdspan_to_view(A);
+
+  Scalar result = {};
+  Kokkos::Max<Scalar> reducer(result);
+  Kokkos::parallel_reduce(Kokkos::RangePolicy(ExeSpace(), 0, A_view.extent(0)),
+			  KOKKOS_LAMBDA (const std::size_t i, Scalar & update)
```

Comment:
[Optional] Would you consider one of the following options?

1. Using `Kokkos::MDRange`,
2. using nested parallelism, or
3. flattening out the parallelism?

For example, Option (3) with iterating over rows would look like this.

```c++
Kokkos::parallel_reduce(
  Kokkos::RangePolicy(ExeSpace(), 0, A_view.extent(0) * A_view.extent(1)),
  KOKKOS_LAMBDA (const std::size_t k, Scalar& update) {
    using ats = Kokkos::Details::ArithTraits<ElementType>;
    const auto i = k / A_view.extent(1);
    const auto j = k % A_view.extent(1);
    reducer.join(update, ats::abs(A_view(i, j)));
  }, reducer);
```

Reply:
yes definitely! My plan was to start with a "first" impl of every function and then go back to improve those impl.

Reply:
is it ok if I make these improvements in a followup PR?

Reply:
done for this!